File: py_backend/flaskr/output.py
import os
import openai
from dotenv import load_dotenv
load_dotenv()

openai.api_key = os.getenv("GPT_API_KEY")

class Output:
    def __init__(self, path):
        self.path = path

    def speak(self, text):
        print(text)

        with openai.audio.speech.with_streaming_response.create(
                model="tts-1",
                voice="alloy",
                input=text
        ) as response:
            response.stream_to_file(self.path)

        print("---saved---")

        print('playing sound using native player')
        os.system("afplay " + self.path)

        print("---played---")

        return True

if __name__ == "__main__":
    path = "../../audio_files/output.mp3"
    output = Output(path)
    output.speak("hello there my name is ethan")

File: js_backend/output.mjs
import OpenAI from 'openai';
import dotenv from 'dotenv';
import player from 'play-sound';
import * as fs from 'fs';
dotenv.config();

const openai = new OpenAI({
  apiKey: process.env['GPT_API_KEY'], // This is the default and can be omitted
});

class Output {
    constructor(filePath) {
        this.filePath = filePath;
        this.player = player();
    }

    async speak(text) {
        console.log('To Speak: ', text);

        const response = await openai.audio.speech.create({
            model: "tts-1",
            voice: "alloy",
            input: text,
        });

        const fileStream = fs.createWriteStream(this.filePath);

        // Promisify the stream events
        await new Promise((resolve, reject) => {
            fileStream.on('finish', resolve);
            fileStream.on('error', reject);
            response.body.pipe(fileStream);
        });

        console.log('File saved at:', this.filePath);

        // Play the audio using the default audio player of your system
        this.player.play(this.filePath);
    }
}

const path = "../audio_files/output.mp3";
const speaker = new Output(path);
await speaker.speak("Hi, this is a test");